### PR TITLE
#380 Introduced a helper method that can be configured in the AR mode…

### PR DESCRIPTION
**Pull request by vraravam**
…l so as to retrieve the associated parent records irrespective of whether the actual parent has been soft-deleted or not.

### CODE REVIEW

**Comment by vraravam** [comment]
@BenMorganIO - any updates on this?

File: lib/paranoia.rb [hunk 1]
```diff
@@ -232,6 +234,7 @@ def self.acts_as_paranoid(options={})
       alias_method :destroy_without_paranoia, :destroy
 
       include Paranoia
+      include ParanoiaHelper
```
Comment (BenMorganIO) [inline lib/paranoia.rb]: Are we able to `extend ParanoiaHelper`? That way we don't need the fancy `ActiveSupport::Concern`.

File: lib/paranoia_helper.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+module ParanoiaHelper
```
Comment (BenMorganIO) [inline lib/paranoia_helper.rb]: I'd probably call this something else. `ParanoiaHelper` isn't very immediate as to its purpose.

File: lib/paranoia_helper.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    #     collection_for_edit :sales_person, nil, name_prefix: :salesmen
+    #     collection_for_edit :sales_person, -> { User.salesmen }, name_prefix: :salesmen   # This is equivalent to the above
+    #   end
+    #
+    # The method will evaluate the scope_block if specified to get all the values for the edit dropdown. If the scope_block is nil,
+    # the method will try to deduce it from the association. Again, if there is a scope defined for the association class with the
+    # same name as option 'name_prefix' or the pluralized version of the associated symbol, then that will be used to fetch all
+    # records for the dropdown
+    #
+    def collection_for_edit(associated_sym, scope_block = nil, options = {})
```
Comment (BenMorganIO) [inline lib/paranoia_helper.rb]: There's a lot going on inside of this method. Given, this codebase is legacy, but with this helper we have a chance to make things more readable. Are we able to use some private methods and reduce the ABC here?

File: lib/paranoia.rb (hunk 1 above)
Comment (vraravam) [reply lib/paranoia.rb]: @BenMorganIO : I can try to implement this, but, given that this gem is anyway dependent on `rails >= 4.0`, would this make sense?

File: lib/paranoia_helper.rb (hunk 3 above)
Comment (vraravam) [reply lib/paranoia_helper.rb]: @BenMorganIO - sure, i can break this down into multiple private methods.

File: lib/paranoia.rb (hunk 1 above)
Comment (BenMorganIO) [reply lib/paranoia.rb]: When I read this line here I thought it was adding instance methods, then I discovered it was doing something to add class methods. It's a simple change, uses way less code, and is more idiomatic over a concern to include class level methods without any macros.

**Comment by vraravam** [comment]
@BenMorganIO - i have made a couple of small changes. I'm not sure how much more the ABC can be reduced - any suggestions?

**Comment by vraravam** [comment]
@BenMorganIO - After a long wait, I have rebased from the upstream branch and would like to get this PR merged. If not, I will close the PR. Please suggest how to go forward.

**Comment by vraravam** [comment]
@BenMorganIO - any decision please?

**Comment by mathieujobin** [comment]
I'm sorry @vraravam we never got back to you after your last update... 
I think your changes actually look good. I don't know if anybody else wanted to use them. That's all.

**Comment by mathieujobin** [comment]
Thanks again for your contribution

**Pull request closed without merging**